src/Chocolatey.PowerShell/Helpers/EnvironmentHelper.cs

```diff
@@ -226,7 +226,8 @@ public static void UpdateSession(PSCmdlet cmdlet)
             var computerName = GetVariable(cmdlet, EnvironmentVariables.System.ComputerName, EnvironmentVariableTarget.Process);
 
             // User scope should override (be checked after) machine scope, but only if we're not running as SYSTEM
-            if (userName != computerName && userName != Shared.EnvironmentNames.System)
+            // System scope has a username of "COMPUTERNAME$"
+            if (userName != computerName && userName != $"{computerName}$")
             {

```

Comment:
We... might wanna confirm if the SYSTEM environment name is used like this code originally expected anywhere (older OSes maybe?), maybe we hould be doing all three checks?